cobspec/src/main/java/com/saralein/cobspec/Main.java

```diff
@@ -14,43 +16,26 @@
 import com.saralein.cobspec.validation.PortValidator;
 import com.saralein.cobspec.validation.Validator;
 import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
 
 public class Main {
     public static void main(String[] args) {
         String home = System.getProperty("user.dir");
-        Logger logger = new ConnectionLogger(System.out);
+        Path logTxt = Paths.get(home + "/log.txt");
+        Logger logger = new AppLogger(System.out, logTxt);
         List<String> validationErrors = runValidationAndReturnErrors(args, home);
 
         if (validationErrors.isEmpty()) {
             ArgsParser argsParser = new ArgsParser(args);
             Integer port = argsParser.parsePort();
             Path root = argsParser.parseRoot(home);
 
-            FormStore formStore = new FormStore();
-            FormBody formBody = new FormBody();
-            FormModification formModification = new FormModification();
-
-            new Application(logger)
-                    .config(new Routes()
-                                .get("/redirect", new RedirectController())
-                                .get("/form", new FormGetController(formStore, formBody))
-                                .post("/form", new FormPostController(formStore, formBody, formModification))
-                                .put("/form", new FormPutController(formStore, formBody, formModification))
-                                .delete("/form", new FormDeleteController(formStore))
-                                .options("/method_options", new OptionsController(Methods.allowNonDestructiveMethods()))
-                                .get("/method_options", new DefaultController())
-                                .put("/method_options", new DefaultController())
-                                .post("/method_options", new DefaultController())
-                                .head("/method_options", new DefaultController())
-                                .options("/method_options2", new OptionsController(Methods.allowGetAndOptions()))
-                                .get("/method_options2", new DefaultController())
-                                .get("/tea", new DefaultController())
-                                .get("/coffee", new CoffeeController()))
-                    .start(port, root);
+            createServer(root, port, logger, logTxt);
         } else {
-            logger.log(String.join("\n", validationErrors));
+            logger.info(String.join("\n", validationErrors));
```

Comment:
if the program is terminating, I'm not sure that "info" would be the correct log level